pkg/executorconfig/v1alpha1/executor_config.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"github.com/ghodss/yaml"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+)
+
+// Resource contains resource details which are going to be updated values
+// of ObjectName, ObjectNamespace, ObjectKind passed to cast at runtime.
+type Resource struct {
+	// ObjectName is name of the resource which is going to be updated
+	// use - { .TaskConfig.ObjectName } to get this value in runtask
+	ObjectName string `yaml:"objectName"`
+	// ObjectNamespace is namespace of the resource which is going to be updated
+	// use - { .TaskConfig.ObjectNamespace } to get this value in runtask
+	ObjectNamespace string `yaml:"objectNamespace"`
+	// ObjectKind is kind of the resource which is going to be updated
+	// use - { .TaskConfig.ObjectKind } to get this value in runtask
+	ObjectKind string `yaml:"objectKind"`
+	// CastemplateName is cast which will be used to update this resource
+	CastemplateName string `yaml:"castemplate"`
+}
+
+// ExecutorConfig provides configuration to all the castemplate for updating mentioned
+// resources values of BaseVersion, TargetVersion, UpgradeTaskCRName, RetryOnError
+// RollbackOnError passed to cast at runtime.
+type ExecutorConfig struct {
```

Comment:
The struct definitions should be in `pkg/apis/openebs.io/upgrade/v1alpha1`

Reply:
will do in next pr.